coffea/hist/hist_tools.py

```diff
@@ -964,13 +964,22 @@ def fill(self, **values):
 
         if self.dense_dim() > 0:
             dense_indices = tuple(d.index(values[d.name]) for d in self._axes if isinstance(d, DenseAxis))
+            xy = np.atleast_1d(np.ravel_multi_index(dense_indices, self._dense_shape))
```

Comment:
A micro-optimisation would be to re-use the dense_indices variable so the input arrays can be deallocated